# **Projeto de Estatística Aplicada a Computação**
## **Aluno:** Vinícius Celestino de Medeiros
## **Matrícula:** 202011250014
## **Professor:** Paulo Ribeiro Lins Junior
## **Data:** 21/02/2021

## **1ª Tarefa**

#### Importando a biblioteca pandas e fazendo a leitura dos arquivos xlsx com a função read_excel.


In [1]:
import pandas as pd
dados_2019 = pd.read_csv("https://raw.githubusercontent.com/vinimedeiros13/Statistic/main/Exercícios/Unidade2/dados1.csv", sep=";", encoding='latin-1')
dados_2020 = pd.read_csv("https://raw.githubusercontent.com/vinimedeiros13/Statistic/main/Projeto/datatran2020.csv", sep=";", encoding='latin-1')

UnicodeEncodeError: 'ascii' codec can't encode character '\xed' in position 40: ordinal not in range(128)

# **2ª Tarefa**

## A)

## **O código abaixo apresenta a quantidade de acidentes que ocorreram no ano de 2019 e 2020:**

#### Função shape utilizada para mostrar a quantidade de linhas que corresponde a quantia de acidentes.

In [2]:
dados_2019.shape[0]

67446

#### Printando a quantidades de acidentes no ano de 2019, que corresponde a quantidade de linhas da planilha;

In [3]:
print("Acidentes em 2019: 67446")

Acidentes em 2019: 67446


#### Função shape utilizada para mostrar a quantidade de linhas que corresponde a quantia de acidentes.

In [4]:
dados_2020.shape[0]

63447

#### Printando a quantidades de acidentes no ano de 2020 (da mesma forma que no código anterior), que corresponde a quantidade de linhas da planilha;


In [5]:
print("Acidentes em 2020: 63447")

Acidentes em 2020: 63447


## **Nota-se uma diminuição de aproximadamente 4000 acidentes de 2019 para 2020.**

## B)

### **Agora é mostrado a quantidade de variáveis que é registrado em cada conjunto de dados:**

In [6]:
# Utilizando uma função info para ver as variáveis no ano de 2019.
dados_2019.shape[1]

30

In [7]:
print("Há 30 colunas de variáveis")

Há 30 colunas de variáveis


In [8]:
# Utilizando uma função info para ver as variáveis no ano de 2019.
dados_2020.shape[1]

30

In [9]:
print("Há 30 colunas de variáveis")

Há 30 colunas de variáveis


## C)

### **Agora, o código a seguir apresenta as cinco cidades brasileiras onde mais ocorreram acidentes em rodovias federais:**

#### Primeiro filtrando os dados pelo róturo 'municipio' utilizando a função values_counts;
#### Função head utilizada para printar as 5 primeiras linhas, que corresponde as 5 cidades onde mais ocorreram acidentes;
#### Variável muni2019 e muni2020 criada para armazenar em DataFrame as 5 primeiras linhas de dados do rótulo 'municipio':

In [10]:
muni2019 = dados_2019['municipio'].value_counts().head().to_frame()
muni2020 = dados_2020['municipio'].value_counts().head().to_frame()

### Resultados obtidos:
#### 5 cidades onde mais ocorreram acidentes em 2019

In [11]:
muni2019

municipio
CURITIBA        1094
BRASILIA        1090
SAO JOSE         819
GUARULHOS        717
SERRA            633

#### 5 cidades onde mais ocorreram acidentes em 2020

In [12]:
muni2020

municipio
BRASILIA        1040
CURITIBA         945
GUARULHOS        639
SAO JOSE         629
PALHOCA          593

## **Brasília assume a liderança, enquanto Curitiba e São José caem de posição, por sua vez, Guarulhos sobe de posição e passa São José.**

## D)

### **Código criado para gerar a quantidade de feridos graves na Paraíba em 2019**

##### Criando a variável "feridos" para puxar os dados do rótulo "feridos_graves" que são maiores ou iguais a 1;

In [13]:
feridos = dados_2019["feridos_graves"] >= 1

##### Agora uma nova variável para pegar os valores obtidos em "feridos";

In [14]:
feridos_ = dados_2019[feridos]

#### Variável "feridospb" responsável por armazenar os dados da Paraíba;

In [15]:
feridospb = feridos_['uf'] == 'PB'

#### Finalizando criando uma nova variável para armazenar os valores anteriores obtidos;

In [16]:
final = feridos_[feridospb]

#### Printando o valor final utilizando a função shape[0] para agrupar a quantidade de linhas no índice 0.

In [17]:
print("Feridos graves na Paraíba:", final.shape[0])

Feridos graves na Paraíba: 414


# **3ª Tarefa**

## A)

### **Agora é criado um novo Dataframe, salvo com a terminação .csv.**
### **Em seguida, é gerado um ranking de acidentes por estados em ambos os anos.**

#### Duas variáveis novas para armazenar os dados do rótulo "uf" (estados);

In [18]:
acidentes_estado2019 = dados_2019["uf"].value_counts()
acidentes_estado2020 = dados_2020["uf"].value_counts()

#### Soma dos dados por estado de 2019 e 2020;

In [19]:
acidentes_total = acidentes_estado2019 + acidentes_estado2020

#### Renomeando a série para "total_acidentes" com a função name;

In [20]:
acidentes_total.name = 'total_acidentes'

#### Filtrando pelo rótulo "uf" com a função index.name;

In [21]:
acidentes_total.index.name = 'uf'

#### Organizando os valores com a função sort_values ...

In [22]:
acidentes_total.sort_values(inplace = True, ascending=True) 

#### Criando uma novo dataframe com uma nova variável chamada acidentestotal_df;

In [23]:
acidentestotal_df = pd.DataFrame(acidentes_total)

#### Salvando o arquivo em csv com a função to_csv;

In [24]:
acidentestotal_df.to_csv("Arquivo.csv")

#### Lendo o arquivo csv com a função read_csv;

In [25]:
acidentestotal = pd.read_csv("Arquivo.csv")

#### Fazendo o rank de acitentes totais com a função rank e usando o method=min para caso haja empate; 

In [26]:
resultados = acidentestotal['total_acidentes'].rank(method="min")

#### Printando a tabela final com o total de acidentes por estado nos anos de 2019 e 2020.

In [27]:
acidentestotal

uf  total_acidentes
0   AM              238
1   AP              283
2   RR              487
3   AC              554
4   TO             1115
5   SE             1186
6   AL             1257
7   PA             1776
8   DF             2130
9   MA             2226
10  PI             2546
11  RN             2770
12  PB             2965
13  MS             2974
14  RO             3018
15  CE             3263
16  MT             4326
17  PE             5293
18  ES             5372
19  GO             6580
20  BA             6914
21  SP             8417
22  RS             8771
23  RJ             8828
24  PR            14876
25  SC            15645
26  MG            17083

## **O estado do Amazonas conteve o menor numéro de acidentes (238), enquanto Minas Gerais obteve o alarmante valor de 17083 acidentes em 2 anos, ficando em primeiro lugar na lista.**

## B)

### **Agora, a função do código a seguir é apresentar os acidentes por dia da semana para os anos de 2019 e 2020.**

#### Duas variáveis criadas para os anos de 2019 e 2020, para filtrar os dados do rótulo "dia_semana" com a função value_counts;

In [28]:
semanas2019 = dados_2019["dia_semana"].value_counts()
semanas2020 = dados_2020["dia_semana"].value_counts()

#### criando uma variável para somar os dados de 2019 e 2020;

In [29]:
semanas = semanas2019 + semanas2020

#### Renomeando a Serie para "acidentes" com a função name";

In [30]:
semanas.name = 'acidentes'

#### Filtrando o rótulo "dia_semana" com a função index.name;

In [31]:
semanas.index.name = 'dia_semana'

#### Organizando os valores com a função sort_values;


In [32]:
semanas.sort_values(inplace = True,ascending=True) 

#### Criando uma novo dataframe com uma nova variável chamada semanas_df;

In [33]:
semanas_df = pd.DataFrame(semanas)

#### Salvando o arquivo em csv com a função to_csv;

In [34]:
semanas_df.to_csv("semanas.xlsx")

#### Lendo o arquivo csv com a função read_csv;

In [35]:
semanastotal = pd.read_csv("semanas.xlsx")

#### Fazendo o rank de acitentes totais com a função rank e usando o method=min para caso haja empate; 

In [36]:
resultados = semanastotal['acidentes'].rank(method="min")

#### Printando a tabela final com o total de acidentes por dias da semana para os anos de 2019 e 2020.

In [37]:
semanastotal

dia_semana  acidentes
0    terça-feira      16177
1   quarta-feira      16516
2   quinta-feira      16981
3  segunda-feira      17638
4    sexta-feira      20131
5        domingo      21706
6         sábado      21744

## **Terça-feira ficou em último lugar na tabela (16177), o principal motivo é por ser dia de semana e o tráfego ser motivado por mobilidade ao trabalho e escolas. O domingo ficou em primeiro lugar com 21744 acidentes, isso se dá pelo fato de nos finais de semana ocorrerem festas ou encontros, consequentemente o uso de bebidas alcoólicas, aumentando o risco de acidentes.**

## **4ª Tarefa**

## **O código a seguir apresenta a causa mais frequente e rara de acidentes registrados em 2019 e 2020.**

## A)

#### Filtrando os dados na coluna "causa acidente em ambos os anos".

In [38]:
freq2019 = dados_2019['causa_acidente'].value_counts() 
freq2020 = dados_2020['causa_acidente'].value_counts()

#### Buscando os 3 primeiros valores encontrados em 2019

In [39]:
print("2019:")
freq2019.head(3).to_frame()

2019:


causa_acidente
Falta de Atenção à Condução                                 24992
Desobediência às normas de trânsito pelo condutor            8098
Velocidade Incompatível                                      6028

#### Buscando os 3 primeiros valores encontrados em 2020

In [40]:
print("2020:")
freq2020.head(3).to_frame()

2020:


causa_acidente
Falta de Atenção à Condução                                 22414
Desobediência às normas de trânsito pelo condutor            8801
Velocidade Incompatível                                      5878

### **Falta de Atenção à Condução como valor mais frequente em 2020 (22414)**

### **Falta de Atenção à Condução como valor mais frequente em 2019 (24992). Isso levanta uma pauta sobre tal causa, celulares, por exemplo, estão como motivo frequente na falta de atenção ao volante, os motoristas perdem o foco na pista para realizar algo no seu smartphone, causando assim um acidente.**

#### Calculando a soma dos valores de 2019 e 2020 da causa de acidentes mais ocorrida.

In [41]:
freqtotal = freq2019["Falta de Atenção à Condução"] + freq2020["Falta de Atenção à Condução"]
print("Nos dois anos somados: ",freqtotal)

Nos dois anos somados:  47406


#### Buscando os 3 últimos valores encontrados em 2019;

In [42]:
freq2019.tail(3).to_frame()

causa_acidente
Fenômenos da Natureza                                          230
Deficiência ou não Acionamento do Sistema de Il...             160
Ingestão de Substâncias Psicoativas                             94

### **Ingestão de Substâncias Psicoativas como o valor mais raro (94).**

#### Buscando os 10 últimos valores encontrados em 2020.

In [43]:
freq2020.tail(10).to_frame()

causa_acidente
Ingestão de álcool ou de substâncias psicoativa...               1
Acostamento em desnível                                          1
Ausência de sinalização                                          1
Acumulo de óleo sobre o pavimento                                1
Condutor usando celular                                          1
Afundamento ou ondulação no pavimento                            1
Trafegar com motocicleta (ou similar) entre as ...               1
Acumulo de água sobre o pavimento                                1
Pedestre cruzava a pista fora da faixa                           1
Frear bruscamente                                                1

### **As 10 causas de acidentes mais raras em 2020.**

## B)

### O código a seguir apresenta a proporção de pessoas ilesas e de feridos graves por mês nos anos de 2019 e 2020

In [44]:
# Criando uma lista dos meses do ano para adicionar aos índices.
meses = ["January", "February", "March","April", "May", "June", "July", "August", "September", "October", "November", "December"]

#### Vítimas ilesas em 2019

In [45]:
ilesos_2019 = dados_2019.groupby(pd.DatetimeIndex(dados_2019.data_inversa).month)["ilesos"].sum()
ilesos2019 = pd.array(ilesos_2019)

#### Ferídos graves em 2019

In [46]:
feridos_2019 = dados_2019.groupby(pd.DatetimeIndex(dados_2019.data_inversa).month)["feridos_graves"].sum()
feridos2019 = pd.array(feridos_2019)

#### Vítimas ilesas em 2020

In [47]:
ilesos_2020 = dados_2020.groupby(pd.DatetimeIndex(dados_2020.data_inversa).month)["ilesos"].sum()
ilesos2020 = pd.array(ilesos_2020)

#### Feridos graves em 2020

In [48]:
feridos_2020 = dados_2020.groupby(pd.DatetimeIndex(dados_2020.data_inversa).month)["feridos_graves"].sum()
feridos2020 = pd.array(feridos_2020)

### Total de envolvidos em acidentes 2019

In [49]:
total_2019 = dados_2019.groupby(pd.DatetimeIndex(dados_2019.data_inversa).month)["pessoas"].sum()
total2019 = pd.array(total_2019)

### Total de envolvidos em acidentes 2020

In [50]:
total_2020 = dados_2020.groupby(pd.DatetimeIndex(dados_2020.data_inversa).month)["pessoas"].sum()
total2020 = pd.array(total_2020)

## Proporção ilesos por total de acidentados 2019

In [51]:
proporcao_ilesos_2019 = ilesos2019/total2019

## Proporção ilesos por total de acidentados 2020

In [52]:
proporcao_ilesos_2020 = ilesos2020/total2020

## Proporção de feridos graves por total de acidentados 2019

In [53]:
proporcao_feridos_2019 = feridos2019/total2019

## Proporção de feridos graves por total de acidentados 2020

In [54]:
proporcao_feridos_2020 = feridos2020/total2020

# Resultados finais

In [55]:
# Criando um dicionário para adicionar todos os dados de ilesos e feridos graves em ambos os anos;
dic = {"Ilesos 2019":proporcao_ilesos_2019,"Ilesos 2020":proporcao_ilesos_2020,"Feridos 2019":proporcao_feridos_2019,"Feridos 2020":proporcao_feridos_2020}
# Transformando os valores em Data Frame e exibindo-os.
totaldedados = pd.DataFrame(dic,index=meses)
totaldedados

Ilesos 2019  Ilesos 2020  Feridos 2019  Feridos 2020
January       0.424224     0.415565      0.113264      0.107463
February      0.424104     0.420011      0.106789      0.111356
March         0.433193     0.401040      0.114599      0.112421
April         0.415685     0.376492      0.105970      0.117780
May           0.415162     0.401271      0.120750      0.115812
June          0.423757     0.423486      0.121617      0.115896
July          0.436815     0.424454      0.116670      0.108501
August        0.413084     0.423510      0.116683      0.118872
September     0.421064     0.416369      0.110656      0.117303
October       0.418922     0.411188      0.119805      0.119475
November      0.425812     0.407667      0.111047      0.120060
December      0.422265     0.412464      0.114405      0.121702

## **Houve sim uma queda na proporção de ilesos por total de feridos em 2020, em 2019 essa taxa estava maior.**
## **Na proporçao de feridos graves por total de feridos há um aumento no ano de 2020 em relação a 2019, em dezembro a taxa subiu de 0.114 para 0.121.**

## C)

In [56]:
dados_2019.shape[0]

67446

In [57]:
dados_2020.shape[0]

63447

## **2019 teve um total de 67446 acidentes, enquanto 2020 teve 63447 acidentes. Houve uma diminuição na quantidade de acidentes, um fator a ser considerado é a pandemia que começou em 2020 no Brasil, fazendo vários brasileiros ficarem em suas casas se protegendo, saindo apenas para o necessário. Logo o tráfego ficou menor, consequentemente os índices de acidentes diminuíram**